src/rgw/rgw_acl_swift.cc

```diff
@@ -276,20 +274,10 @@ void RGWAccessControlPolicy_SWIFTAcct::add_grants(const DoutPrefixProvider *dpp,
 
     if (uid_is_public(uid)) {
       grant.set_group(ACL_GROUP_ALL_USERS, perm);
-      acl.add_grant(grant);
     } else  {
-      std::unique_ptr<rgw::sal::User> user = driver->get_user(rgw_user(uid));
-
-      if (user->load_user(dpp, null_yield) < 0) {
-        ldpp_dout(dpp, 10) << "grant user does not exist:" << uid << dendl;
-        /* skipping silently */
-        grant.set_canon(user->get_id(), std::string(), perm);
-        acl.add_grant(grant);
-      } else {
-        grant.set_canon(user->get_id(), user->get_display_name(), perm);
-        acl.add_grant(grant);
-      }
+      grant = user_to_grant(dpp, driver, uid, perm);
```

Comment:
is this a new helper function?

Reply:
it's not new, no. using it just avoided some code duplication here